cloud/azure/actuators/cluster/clusteractuator.go

```diff
@@ -64,6 +65,12 @@ func (azure *AzureClusterClient) Reconcile(cluster *clusterv1.Cluster) error {
 	if err != nil {
 		return fmt.Errorf("error loading cluster provider config: %v", err)
 	}
+
+	_, err = azure.resourcemanagement().CreateOrUpdateGroup(clusterConfig.ResourceGroup, clusterConfig.Location)
```

Comment:
the resource group creation and updating should be part of the cluster reconciliation loop